### PR TITLE
Fix typos, casing, md in CHANGES.md

### PR DESCRIPTION
**Pull request by charlie-cadmv**
Fix casing on proper names and some typos.
Also fixes some markdown syntax errors, adds proper indentation to bulleted lists, and to marks up keywords like method names outside of code blocks.
Small fix to a code example under [Changes in 1.3. beta 2](https://github.com/charlie-cadmv/beanshell/blob/master/CHANGES.md#changes-in-13-beta-2) to add '//' before comment text.

### CODE REVIEW

**Comment by charlie-cadmv** [comment]
Sorry there's so much to review. If you have any suggestions on how I can do things differently to make a change like this easier to review I'm all ears. I spend a lot of my time doing code reviews (not usually Markdown though).

File: CHANGES.md [hunk 1]
```diff
@@ -98,7 +98,7 @@ construction (instance creation... along with any constructor).
 ### Added JSR223 javax.script adapter engine implementation in a new bsh.engine package
 
 This implementation of the spec supports pluggable ScriptContext.  Please
-see the TestBshScriptEngine.java file (and the spec) for a starting point
+see the `TestBshScriptEngine.java` file (and the spec) for a starting point
```
Comment (Sailsman63) [inline CHANGES.md:101]: Might want to link to the actual file. Using a relative link like so: `[TestBshScriptEngine.java](src/test/java/bsh/TestBshScriptEngine.java)`

File: CHANGES.md [hunk 2]
```diff
@@ -132,20 +130,17 @@ Added test case.
...
-    optionally, thread context classloader
-
-    plain Class.forName()
-    source class (.java file in classpath)
+    * in-script evaluated class (scripted class)
+    * in-script added / modified classpath
+    * optionally, external classloader
+    * optionally, thread context classloader
+    * plain `Class.forName()`
+    * source class (.java file in classpath)
```
Comment (Sailsman63) [inline CHANGES.md:138]: Indenting lists is not really necessary. Whether they are easier to read like this is open to discussion.

File: CHANGES.md [hunk 3]
```diff
@@ -154,36 +149,35 @@ now run in untrusted environment, with limitations.  Applets cannot
...
-```java
-print( myObject instanceof Blah );
-print( myObject.getFoo() );
-print( myObject.somethingElse() );
-```
+    ```java
+    print( myObject instanceof Blah );
+    print( myObject.getFoo() );
+    print( myObject.somethingElse() );
+    ```
```
Comment (Sailsman63) [inline CHANGES.md:158]: Indenting fenced code blocks is *really* not necessary, and IMO harder to maintain when editing the file.

Might also want a blank line before and after the fenced block for readability of the raw markdown.

Noticed this throughout the file.

**Review by Sailsman63 — COMMENTED** [review]
I end up doing a lot of review on the projects I'm involved in, as well. I've found that the best way to improve reviewability, at least from my standpoint, is to get the contributor to break their PR up into logical steps: You want to refactor some code so that your feature is simpler to integrate/does not duplicate code? Fine. Do the refactor in a separate commit *before* the feature, so I can see it in isolation.

That said, I don't think that there would be a good way to have broken these up.

Caveat: I'm not really a maintainer here. These comments are my own opinion, and may be wrong.

**Comment by nickl-** [comment]
The typo fixes are welcome but this is not exactly up to date by any degree. It is a start I guess.

@Sailsman63 added valid review comments (thank you) which should be addressed first, then we can merge this.

@charlie-cadmv please apply fixes if you get a moment.

File: CHANGES.md (hunk 2 above)
Comment (charlie-cadmv) [reply CHANGES.md:138]: I interpreted the original text as a list within a parent list element. Indentation is required to render it that way in markdown.
Here's what I think we want:
image

Here's how it looks with indentation removed, which has a very different meaning:

image

File: CHANGES.md (hunk 3 above)
Comment (charlie-cadmv) [reply CHANGES.md:158]: This code block is part of a bulleted list element. All content belonging to a list element must also be indented to the same level as the first line of the bulleted element. If you don't indent it, it's no longer part of the list, which changes how markdown renders it (at least in GitHub).

File: CHANGES.md (hunk 2 above)
Comment (Sailsman63) [reply CHANGES.md:138]: Good point. Somehow I missed this.

File: CHANGES.md (hunk 3 above)
Comment (Sailsman63) [reply CHANGES.md:158]: Also fair. somehow, these *look* like they are indented further than the list-element, though.

File: CHANGES.md (hunk 1 above)
Comment (charlie-cadmv) [reply CHANGES.md]: Good idea, I have linkified all of the references to .java files.

File: CHANGES.md (hunk 3 above)
Comment (charlie-cadmv) [reply CHANGES.md]: The code block and text in this list item were not all evenly indented. Fixed here: [6c9610c](https://github.com/beanshell/beanshell/pull/641/commits/6c9610c84923d3e87b178f6c54b0ec1ef4aa7994)

This doesn't affect rendering but does make it easier to read the raw Markdown.

Comment (charlie-cadmv) [reply CHANGES.md]: @Sailsman63 Did my last change to the indentation ([6c9610c](https://github.com/beanshell/beanshell/pull/641/commits/6c9610c84923d3e87b178f6c54b0ec1ef4aa7994)) fix the indentation issue you saw?

**Comment by charlie-cadmv** [comment]
@nickl- I believe my updates have addressed @Sailsman63's concerns with this PR and that it is ready to merge.

File: CHANGES.md (hunk 3 above)
Comment (Sailsman63) [reply CHANGES.md]: Sorry, I meant to reply earlier. Yes, this looks good.

**Comment by nickl-** [comment]
> @nickl- I believe my updates have addressed @Sailsman63's concerns with this PR and that it is ready to merge.

Thank you @charlie-cadmv will make some time to review again.

**Comment by nickl-** [comment]
@jimjag thanks for the assist

**Pull request merged**